modules/common/deployment/deployment.go

```diff
@@ -34,11 +34,11 @@ import (
 // NewDeployment returns an initialized Deployment.
 func NewDeployment(
 	deployment *appsv1.Deployment,
-	timeout int,
+	timeout time.Duration,
 ) *Deployment {
 	return &Deployment{
 		deployment: deployment,
-		timeout:    time.Duration(timeout) * time.Second,
+		timeout:    timeout,
```

Comment:
you can drop `SetTimeout` below. It was a stop gap solution to re-define smaller than 1 second timeout during test without breaking the `NewDeployment` interface. But now that we are changing that interface the timeout can be defined directly in `NewDeployment()` with less than 1 sec granularity.  I think it is only used by nova-operator.

Reply:
Thanks for sharing that backgroupd. Now this replacement would make more sense :-)